### PR TITLE
feat: added icon to show the current status of adapter deployement

### LINKED ISSUE #8276: [UI] Show current status of deployment or undeployment for adapters extension
#### Current Behavior

For adapters extensions as of now to deploy or undeploy we toggle the switch and since the deployment and undeployment takes some time. There is  no feedback for what is really happening. All we know is that the operation is currently in progress.

#### Desired Behavior

It would be great if we could show a status chip and icon state for our adapter.

We can have the status icon like the below one (different color for different status)  :-

image

The status could be one of 4 things:-
- Deploying (yellow)
- Undeploying (orange)
- Connected  (green)
- Disconnected (gray or red)

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Yana-Gupta**
**Notes for Reviewers**

This PR fixes #8276

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Yana-Gupta** [comment]
[Screencast from 24-08-23 04:58:26 PM IST.webm](https://github.com/meshery/meshery/assets/103108630/3bdd98e0-d309-4a39-971c-3e6eb3352449)

**Comment by Yana-Gupta** [comment]
> @Yana-Gupta, please add this indicator to the adapter chips under Settings --> Adapters. Use the same green and pulsation that is used to indicate status on the Kubernetes chips. Screenshot023-08-24 at 4 29 49 PM
> 
> Pulsating
> 
>  Screen.Recording.2023-08-24.at.4.29.24.PM.mov

Okay

**Comment by Yana-Gupta** [comment]
[Screencast from 29-08-23 11:38:19 AM IST.webm](https://github.com/meshery/meshery/assets/103108630/0564323b-516f-4e46-8f27-44d62fd84fa9)


@leecalcote please review.

I am presuming that when an adapter is initially displayed within the chip as the page is rendered, it indicates its deployment.

**Comment by Yana-Gupta** [comment]
Current State 

[Screencast from 01-09-23 02:41:43 PM IST.webm](https://github.com/meshery/meshery/assets/103108630/ed558bea-696f-4468-a789-159cb5a585fd)

File: ui/components/MeshAdapterConfigComponent.js [hunk 1]
```diff
@@ -138,6 +152,32 @@ class  MeshAdapterConfigComponent extends React.Component {
...
+      const lable = adapter.name.toUpperCase()
+      initialAdapterStates[lable] = STATUS.UNDEPLOYED;
+    });
+
+    this.setState({
+      adapterStates : initialAdapterStates,
+    });
+  }
+
+  getStatusColor = (status) => {
```
Comment (leecalcote) [inline ui/components/MeshAdapterConfigComponent.js:169]: Some pretty good stuff here, @Yana-Gupta. 🙂

Comment (leecalcote) [reply ui/components/MeshAdapterConfigComponent.js:169]: We'll want to further centralize this states and their colors after this PR merges.

WE're about to start on a big journey into Connections and management of objects that MeshSync discovers.

Comment (Yana-Gupta) [reply ui/components/MeshAdapterConfigComponent.js:169]: Thanks! I am glad to hear about the plan to centralise the states and eager to contribute and help with the upcoming journey.

**Pull request merged**